papers/MolCLR_2024.md

```diff
@@ -0,0 +1,101 @@
+---
+author: Sreekanth Kunchapu
+date: 2024-05-29
+bibliography: ./references_MolCLR.bib
+---
+
+# Molecular contrastive learning of representations (MolCLR) via graph neural networks
+
+## Why discussing this paper? 
+
+I chose Wang's et al.'s paper [@wang2022molecular] for our journal club because
+
+- Labelled datasets are hard to acquire since conducting experiments is expensive and time-consuming.
+- Labelled datasets are limited, if we train the model on the limited datasets, we might overfit the models and does not generalizable.
+- The authors propose a framework to learn better representations with unlabelled datasets during pretraining, the pretained model is plugged in for the downstream tasks.
+
+## Context
+
+The paper introduces MolCLR, a self-supervised learning framework that uses graph encoders to enhance molecular representations effectively.
+
+They considered a large unlabelled dataset with 10 million unique molecules SMILES from ChemBERTa and PubChem. This framework has two important steps. First, in the pre-training phase, they build molecule graphs and develop graph neural network encoders to learn differentiable representations. Second, the pretrained GNN backbone is used for the supervised learning tasks.
+
+## Main idea behind contrastive learning
+The aim of contrastive representation learning is to develop an embedding space where similar samples are positioned closely together, whereas dissimilar samples are kept distant from each other.
+
+![Figure taken from the blog post by Ekin Tiu illustrating the idea behind contrastive learning.](MolCLR_2024_images/Constrative learning .png)
+
+## NT-Xent loss
+
+The authors use a NT-Xent loss to train the graph encoders.
+
+$$
+L_{ij} = -\log \left( \frac{\exp(\text{sim}(z_i, z_j)/\tau)}{\sum_{k=1}^{2N} \mathbf{1}_{\{k \neq i\}} \exp(\text{sim}(z_i, z_k)/\tau)} \right)
+$$
+
+- `z_i` and `z_j`: Latent vectors extracted from a positive data pair.
+- `N`: Batch size, indicating the number of data pairs used.
+- `sim(⋅)`: Function measuring the similarity between two vectors.
+- `τ`: The temperature parameter, used to scale the similarity measures in the function.
+
+To get more understanding, there is literature paper on contrastive learning one shoould read [@le2020contrastive].
+
+## Overview of MolCLR framework
+
+- The SMILES are converted into graphs in a batch. The graphs consists of nodes and edges. Nodes represents atoms and edges represents chemical bonds of a molecule.
+
+- The authors studied three types of augmentations like atom masking, bond deletion and sub-graph removal.
+
+- In the current work, they used sub-removal with some probability 0.25. Now, the model tries to learn hidden patterns of the molecule within the remaining subgraphs.
```

Comment:
```suggestion
- In the current work, they used sub-removal with probability 0.25. Now, the model tries to learn the hidden patterns of the molecule within the remaining subgraphs.
```

what are "hidden patterns"?

Reply:
why is this its own point and other augmentations do not have their own point? That is, why isn't this point simply merged in with the other augmentations?

Reply:
> why is this its own point and other augmentations do not have their own point? That is, why isn't this point simply merged in with the other augmentations?

They did some analysis on augmentations rights, out of all augmentations sub-graph removal with P=0.25 has achieved good perfoemance. 
the  sub-graph removal is combination of atom masking and bond deletion .

Reply:
Also here I tried to highlight something that would perhaps be good to clarify in the text because it is not clear to the reader of your document.